scripts/showcase/samplesheet-prep-test.py

```diff
@@ -68,9 +68,26 @@ def get_args():
     parser = argparse.ArgumentParser()
 
     # Place arguments
-    parser.add_argument("--samplesheet", "--sample-sheet", "-s", type=str, required=True, dest="sample_sheet")
-    parser.add_argument("--trackingsheet", "--tracking-sheet", "-t", type=str, required=True, dest="tracking_sheet")
-    parser.add_argument("--outputPath", "--output-path", "-o", type=str, required=True, dest="output_path")
+    parser.add_argument("--sample-sheet", "--samplesheet", "-s", type=str, required=True, dest="sample_sheet")
+    parser.add_argument("--tracking-sheet", "--trackingsheet", "-t", type=str, required=True, dest="tracking_sheet")
+    parser.add_argument("--output-path", "--outputPath", "-o", type=str, required=True, dest="output_path")
+
+    sample_params_args = parser.add_argument_group("Parameters for filtering out samples."
+                                                   "If --sample-type, --index-legnth or --index2length"
+                                                   "are used. A file named SampleSheet.csv will be output."
+                                                   "If the --all parameter is used, then each file named will be "
+                                                   "distinguished with the following notation"
+                                                   "SampleSheet_<sample_type>.<index-length>.<index2-length>.csv")
```

Comment:
If I understand this correctly, we had agreed to continue generating `SampleSheet.csv` (which by default would be of type WGS and index length (8,8) - if these indexes exist or (8,6) otherwise? For this case, I'll need to update CWL tool definition by adding ` --sample-type, --index-legnth & --index2length` parameters. If I set a default value for these and not pass any input parameter at workflow launch stage - would it still default to generating `SampleSheet.csv`?

Reply:
Defaults of `sample-type`, `index-length` and `index2length` are `WGS`, `8` and `8`, so you don't have to set any defaults as they're set by the script although you may wish to so that CWL users can see them easier.

If you want to extract the samplesheet for WGS samples with 8 and 6 you would need to set `--index2-length` to '2'
If you don't, this will terminate with exit code 1 since no samples would be found (if no samples have the index 8, 8)

One main change is that it will generate the file `Samplesheet.csv` rather than `SampleSheet_<sample_type>.csv`

Reply:
- `--index-legnth` - typo needs fixing
- both index argument names need to be harmonised